Lib/test/test_zipfile/test_core.py

```diff
@@ -1877,6 +1878,54 @@ def test_read_after_write_unicode_filenames(self):
             zipfp.writestr('приклад', b'sample')
             self.assertEqual(zipfp.read('приклад'), b'sample')
 
+    def create_zipfile_with_extended_timestamp_extra_data(self, filename, extra_timestamp_data):
+        with zipfile.ZipFile(TESTFN, mode='w') as zf:
+            filename_encoded = filename.encode("utf-8")
+            zip_info = zipfile.ZipInfo(filename)
+
+            tag_for_extra_block = b'\x55\x54'
+
+            # Set only modification time
+            flags = b'\x01'
+
+            # modification time
+            if extra_timestamp_data is not None:
+                mtime = struct.pack('<L', extra_timestamp_data)
+            else:
+                mtime = b''
+            extra_data = flags + mtime
+
+            tsize = len(extra_data).to_bytes(2, 'little')
+
+            zip_info.extra = tag_for_extra_block + tsize + extra_data
+
+            # add the file to the ZIP archive
+            zf.writestr(zip_info, b'Hello World!')
+
+    def test_read_zipfile_containing_extended_timestamp_extra_field(self):
+        timestamp_data = int(datetime.datetime(2024, 6, 26, 9, 27, 30).timestamp())
+        self.create_zipfile_with_extended_timestamp_extra_data("aayush.txt", timestamp_data)
+        with zipfile.ZipFile(TESTFN, "r") as zf:
+            self.assertEqual(zf.filelist[0].extended_mtime, timestamp_data)
+
+    def test_read_zipfile_containing_timestamp_beyond_2038(self):
+        # Test for 2038 problem
+        timestamp_data = int(datetime.datetime(2038, 6, 26, 9, 27, 30).timestamp())
+        self.create_zipfile_with_extended_timestamp_extra_data("aayush.txt", timestamp_data)
+        with zipfile.ZipFile(TESTFN, "r") as zf:
+            self.assertEqual(datetime.datetime.fromtimestamp(zf.filelist[0].extended_mtime).year, 1902)
```

Comment:
`fromtimestamp` uses the OS libraries to return a datetime from a timestamp. Unix implementations can handle negative timestamps but windows will throw an error `OSError: [Errno 22] Invalid argument`

For this check, we can just check that the value is negative without converting to a time.

```suggestion
        timestamp_data = -1
        self.create_zipfile_with_extended_timestamp_extra_data("aayush.txt", timestamp_data)
        with zipfile.ZipFile(TESTFN, "r") as zf:
            self.assertEqual(zf.filelist[0].extended_mtime, timestamp_data)
```